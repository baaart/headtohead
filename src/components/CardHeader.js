import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const Flag = styled.img`
  height: 50px;
`
const Wrapper = styled.div`
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
`

function CardHeader({...props}) {
  return (
    <Wrapper>
      <h3>{props.title}</h3>
      <Flag src={props.url}/>
    </Wrapper>
  );
}

CardHeader.propTypes = {
  title: PropTypes.string.isRequired,
  src: PropTypes.string.isRequired
};

export default CardHeader;
