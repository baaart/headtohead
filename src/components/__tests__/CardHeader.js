import React from 'react';
import expect from 'expect';
import { shallow } from 'enzyme';
import CardHeader from '../CardHeader';
import { fakeData } from '../__mocks__/fakeData';

const player = fakeData.players[0];

describe('CardHeader', () => {
  it('Matches snapshot', () => {
    const wrapped = shallow(<CardHeader title={player.firstname} url={player.country.picture} />);
    expect(wrapped).toMatchSnapshot();
  })
});
