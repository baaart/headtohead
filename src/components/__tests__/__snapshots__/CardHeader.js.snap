// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardHeader Matches snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CardHeader
    title="Stan"
    url="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <h3>
          Stan
        </h3>,
        <ForwardRef
          src="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Stan",
        },
        "ref": null,
        "rendered": "Stan",
        "type": "h3",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "src": "https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png",
        },
        "ref": null,
        "rendered": null,
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "rules": Array [
              "
  height: 50px;
",
            ],
          },
          "displayName": "styled.img",
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "img",
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "sc-bwzfXH",
        "isStatic": true,
        "rules": Array [
          "
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
",
        ],
      },
      "displayName": "styled.div",
      "render": [Function],
      "styledComponentId": "sc-bwzfXH",
      "target": "div",
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <h3>
            Stan
          </h3>,
          <ForwardRef
            src="https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Stan",
          },
          "ref": null,
          "rendered": "Stan",
          "type": "h3",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "src": "https://i.eurosport.com/_iss_/geo/country/flag/large/2213.png",
          },
          "ref": null,
          "rendered": null,
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "sc-bdVaJa",
              "isStatic": true,
              "rules": Array [
                "
  height: 50px;
",
              ],
            },
            "displayName": "styled.img",
            "render": [Function],
            "styledComponentId": "sc-bdVaJa",
            "target": "img",
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "sc-bwzfXH",
          "isStatic": true,
          "rules": Array [
            "
  width: 100%;
  display: flex;
  flex-direction: row;
  justify-content: space-between;
",
          ],
        },
        "displayName": "styled.div",
        "render": [Function],
        "styledComponentId": "sc-bwzfXH",
        "target": "div",
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
